src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/ConstructorParametersShouldMatchPropertyNames.cs

```diff
@@ -0,0 +1,194 @@
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterCompilationStartAction(
+                (compilationStartContext) =>
+                {
+                    INamedTypeSymbol? jsonConstructorAttributeNamedSymbol = compilationStartContext.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemTextJsonSerializationJsonConstructorAttribute);
+                    if (jsonConstructorAttributeNamedSymbol == null)
+                    {
+                        return;
+                    }
+
+                    var paramAnalyzer = new ParameterAnalyzer(jsonConstructorAttributeNamedSymbol);
+
+                    compilationStartContext.RegisterOperationBlockStartAction(
+                        (startOperationBlockContext) =>
+                        {
+                            if (startOperationBlockContext.OwningSymbol is IMethodSymbol method
+                                && !paramAnalyzer.ShouldAnalyzeMethod(method))
+                            {
+                                return;
+                            }
+
+                            startOperationBlockContext.RegisterOperationAction(
+                                context => ParameterAnalyzer.AnalyzeOperationAndReport(context),
+                                OperationKind.ParameterReference);
+                        });
+                });
+        }
+
+        private sealed class ParameterAnalyzer
+        {
+            private readonly INamedTypeSymbol _jsonConstructorAttributeInfoType;
+
+            public ParameterAnalyzer(INamedTypeSymbol jsonConstructorAttributeInfoType)
+            {
+                _jsonConstructorAttributeInfoType = jsonConstructorAttributeInfoType;
+            }
+
+            public static void AnalyzeOperationAndReport(OperationAnalysisContext context)
+            {
+                var operation = (IParameterReferenceOperation)context.Operation;
+
+                if (operation.Parent is not IAssignmentOperation assignment)
+                {
+                    return;
+                }
+
+                IParameterSymbol param = operation.Parameter;
+                ISymbol? referencedSymbol = assignment.Target.GetReferencedMemberOrLocalOrParameter();
+
+                if (referencedSymbol == null)
+                {
+                    return;
+                }
+
+                var field = referencedSymbol as IFieldSymbol;
+                var prop = referencedSymbol as IPropertySymbol;
+
+                if (field == null && prop == null)
+                {
+                    return;
+                }
+
+                // Only process instance fields and properties
+                if (referencedSymbol.IsStatic)
+                {
+                    return;
+                }
```

Comment:
This can be combined with the null check above as follows `if (referencedSymbol { IsStatic: false })`

Reply:
Done.